pkgs/misc/uboot/default.nix

```diff
@@ -435,6 +412,31 @@ in {
     defconfig = "rpi_4_defconfig";
     extraMeta.platforms = ["aarch64-linux"];
     filesToInstall = ["u-boot.bin"];
+    extraConfig = ''
+      CONFIG_CMD_NVME=y
+      CONFIG_NVME=y
+      CONFIG_NVME_PCI=y
+      CONFIG_USB_STORAGE=y
+      CONFIG_USB_FUNCTION_MASS_STORAGE=y
+      CONFIG_USB_EHCI_HCD=y
+      CONFIG_USB_EHCI_GENERIC=y
+      CONFIG_USB_OHCI_HCD=y
+
+      CONFIG_USB_XHCI_BRCM=y
+    '';
+    # These were taken from:
+    # https://github.com/home-assistant/operating-system/tree/dev/buildroot-external/board/raspberrypi/patches/uboot
+    # https://github.com/home-assistant/operating-system/tree/dev/buildroot-external/board/raspberrypi/yellow/patches/uboot
+    extraPatches = [
+      ./rpi-cm4/0001-rpi-add-NVMe-to-boot-order.patch
+      ./rpi-cm4/0002-Revert-nvme-Correct-the-prps-per-page-calculation-me.patch
+      ./rpi-cm4/0003-usb-xhci-brcm-Make-driver-compatible-with-downstream.patch
+      ./rpi-cm4/0004-drivers-bcm283x-don-t-explicitly-disable-init.patch
+      ./rpi-cm4/0005-drivers-bcm283x-allow-to-spawn-any-PL011-UART.patch
+      ./rpi-cm4/0006-nvme-improve-readability-of-nvme_setup_prps.patch
+      ./rpi-cm4/0007-nvme-Use-pointer-for-CPU-addressed-buffers.patch
+      ./rpi-cm4/0008-nvme-translate-virtual-addresses-into-the-bus-s-addr.patch
+    ];
```

Comment:
That won't do, NixOS is shipping mainline or "accepted-to-be-mainlined" U-Boot. By "accepted-to-be-mainlined" I mean patches that are already in master and cleanly applies on the latest version.

With that said, it's not like I'm a purist in all forms. The rebased patch set would be acceptable in Tow-Boot, where *functionality* is preferred over staying true to stock. (Hinting at a way forward for the fixes.)

If/when they end-up accepted, the story would be different, and we would refer to the upstream `mbox` or git revisions.

I guess we should comb through the more recent patches see if there's a new alternative written-up to fix the underlying problem.

Reply:
ACK'd removed from this series.